util.cpp

```diff
@@ -0,0 +1,23 @@
+#include "util.h"
+
+#include <cassert>
+
+uint64_t shift(int num_bits)
+{
+    int x = 0;
+    for (int i = 0; i < num_bits; ++i) {
+        x += 1;
+    }
+    assert(x > 0);
+    return static_cast<uint64_t>(1) << (64 - x);
+}
+
+int divide(int length)
```

Comment:
I think we don't need this (?)

Reply:
True. That is an implementation of the function mentioned on the clang analyze website. It shouldn't make a difference, but another very similar bug.

Reply:
Let's focus on one bug, I think it will be quite enough. I can create another branch though.